File: frontend/src/services/types/login-record.ts
import { PaginationParams, PaginationResult } from './common';

// 登录记录相关类型
export interface LoginRecordResponse {
  id: number;
  user_id: number;
  username: string;
  ip_address: string;
  user_agent: string;
  login_time: string;
  logout_time: string | null;
  is_active: boolean;
}

export interface LoginRecordQueryParams extends PaginationParams {
  username?: string;
  ip_address?: string;
  start_time?: string;
  end_time?: string;
  search?: string;
}

export interface MyLoginRecordQueryParams extends PaginationParams {
  ip_address?: string;
  start_time?: string;
  end_time?: string;
  search?: string;
}

// 登录记录分页结果接口（使用records字段）
export interface PaginatedLoginRecords {
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
  records: LoginRecordResponse[];
}

export interface LoginStatisticsSummary {
  total_logins: number;
  active_users: number;
  distinct_ips: number;
  today_logins: number;
  analysis_period_days: number;
}

File: backend/schemas/account/login_record.py
"""
用户登录记录相关的Pydantic模型
"""
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime


class LoginRecordResponse(BaseModel):
    """登录记录响应模型"""
    id: int
    user_id: int
    username: str
    ip_address: str
    user_agent: str
    login_time: datetime
    logout_time: Optional[datetime]
    is_active: bool


class PaginatedLoginRecords(BaseModel):
    """分页登录记录响应"""
    total: int
    page: int
    page_size: int
    total_pages: int
    records: List[LoginRecordResponse]